content/docs/tasks/traffic-management/secure-ingress-sds/index.md

```diff
@@ -0,0 +1,473 @@
+
+1.  Enable SDS at ingress gateway and deploy the ingress gateway agent.
+    This feature is disabled by default, you need to enable the feature flag [`istio-ingressgateway.sds.enabled`]({{<github_blob>}}/install/kubernetes/helm/subcharts/gateways/values.yaml) in helm,
+    and then generate the `istio-ingressgateway.yaml` file:
+
+    {{< text bash >}}
+    $ helm template install/kubernetes/helm/istio/ --name istio --namespace istio-system -x charts/gateways/templates/deployment.yaml \
+    --set gateways.istio-egressgateway.enabled=false --set gateways.istio-ingressgateway.sds.enabled=true > $HOME/istio-ingressgateway.yaml
+    $ kubectl apply -f $HOME/istio-ingressgateway.yaml
+    {{< /text >}}
+
+1.  Set the environment variables `INGRESS_HOST` and `SECURE_INGRESS_PORT`
+
+    {{< text bash >}}
+    $ export SECURE_INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].port}')
+    $ export INGRESS_HOST=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    {{< /text >}}
+
+1.  Start the `httpbin` sample
+
+    {{< text bash >}}
+    $ cat <<EOF | kubectl apply -f -
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: httpbin
+      labels:
+        app: httpbin
+    spec:
+      ports:
+      - name: http
+        port: 8000
+      selector:
+        app: httpbin
+    ---
+    apiVersion: extensions/v1beta1
+    kind: Deployment
+    metadata:
+      name: httpbin
+    spec:
+      replicas: 1
+      template:
+        metadata:
+          labels:
+            app: httpbin
+            version: v1
+        spec:
+          containers:
+          - image: docker.io/citizenstig/httpbin
+            imagePullPolicy: IfNotPresent
+            name: httpbin
+            ports:
+            - containerPort: 8000
+    EOF
+    {{< /text >}}
+
+1.  Create a secret for ingress gateway
+
+    {{< text bash >}}
+    $ kubectl create -n istio-system secret generic httpbin-credential --from-file=key=httpbin.example.com/3_application/private/httpbin.example.com.key.pem --from-file=cert=httpbin.example.com/3_application/certs/httpbin.example.com.cert.pem
+    {{< /text >}}
+
+1.  Define a Gateway with a server section for port 443, and specify `credentialName` to be `httpbin-credential`, which should be the same as the secret name.
+    TLS mode should be specified as SIMPLE. `serverCertificate` and `privateKey` should not be empty.
+
+    {{< text bash >}}
+    $ cat <<EOF | kubectl apply -f -
+    apiVersion: networking.istio.io/v1alpha3
+    kind: Gateway
+    metadata:
+      name: mygateway
+    spec:
+      selector:
+        istio: ingressgateway # use istio default ingress gateway
+      servers:
+      - port:
+          number: 443
+          name: https
+          protocol: HTTPS
+        tls:
+          mode: SIMPLE
+          serverCertificate: "use sds"         # arbitrary non-empty string
```

Comment:
please dont do this.. the intent of the credentialName was that it will be additive. If you put this in tutorial, people will assume that SDS is mutually exclusive with non SDS. Similar to other gateway configs, use the proper cert path. And then state in the explanation that if the operator is sure that SDS is present, then the secrets do not have to be mounted in the path. But if SDS is not present, then these paths will be used as the fallback option, with the assumption that the certs are mounted..

Reply:
Done.